addons/stock/models/stock_picking.py

```diff
@@ -900,7 +900,10 @@ def _create_lots_for_picking(self):
         Lot = self.env['stock.production.lot']
         for pack_op_lot in self.mapped('pack_operation_ids').mapped('pack_lot_ids'):
             if not pack_op_lot.lot_id:
-                lot = Lot.create({'name': pack_op_lot.lot_name, 'product_id': pack_op_lot.operation_id.product_id.id})
+                lot_values = {'name': pack_op_lot.lot_name, 'product_id': pack_op_lot.operation_id.product_id.id}
```

Comment:
life_date field is introduced from product_expiry module.

Please do not write stuff regarding life_date field stuff